src/anemoi/training/diagnostics/mlflow/logger.py

```diff
@@ -492,17 +510,34 @@ def log_hyperparams(self, params: dict[str, Any] | Namespace) -> None:
             if config := params.get("config"):
                 params["config"] = map_config_to_primitives(config)
 
-            params = _flatten_dict(params, delimiter=".")  # Flatten dict with '.' to not break API queries
-            params = self._clean_params(params)
-
             import mlflow
             from mlflow.entities import Param
 
-            # Truncate parameter values.
             truncation_length = 250
+
             if Version(mlflow.VERSION) >= Version("1.28.0"):
                 truncation_length = 500
-            params_list = [Param(key=k, value=str(v)[:truncation_length]) for k, v in params.items()]
+
+            self.log_hyperparams_as_artifact(expand_iterables(params, size_threshold=truncation_length, delimiter="."))
+
+            expanded_params = {}
+            params = dict(params)
+
+            for key in expand_keys or []:
+                if key in params:
```

Comment:
Could we call 'self.log_hyperparams_as_artifact' over the expanded params to avoid calling the expand_iterables twice?

Reply:
It is a question of whether we want to the expand config or not, currently it expands all, rather than just those in expand_keys.